oslogmerger/oslogmerger.py

```diff
@@ -91,39 +152,180 @@ def log_entries(self):
                  level, rest]).rstrip('\n'))
 
 
+def get_path_and_alias(filename, log_base, log_postfix):
+    # check if filename has an alias for log output, in the form of
+    # /path/filename:alias
+    filename_and_alias = filename.rsplit(':', 1)
+    filename = log_base + filename_and_alias[0] + log_postfix
+    alias = filename_and_alias[1:]
+
+    is_url = False
+    if filename == 'http' and alias and alias[0].startswith('//'):
+        filename = filename_and_alias[0] + ':' + filename_and_alias[1]
+        alias = filename_and_alias[2:]
+        is_url = True
+
+    return filename, alias[0] if alias else None, is_url
+
+
+# If we are in level 3, we must reduce all paths to the minimum
+def reduce_strings(strings):
+    num_strs = len(strings)
+    if num_strs == 1:
+        return {strings[0]: strings[0][-1]}
+
+    # Convert each string to a list
+    str_list = [list(s) for s in strings]
+    result = [s.pop() for s in str_list]
+
+    while len(set(result)) != num_strs:
+        for i, s in enumerate(str_list):
+            if s:
+                letter = s.pop()
+                str_list[i].insert(0, letter)
+                result[i] = letter + result[i]
+    result = [''.join(r) for r in result]
+    return {strings[i]: s for i, s in enumerate(result)}
+
+
+def reduce_tree(tree):
+    if not len(tree):
+        return tree
+    reduced = reduce_strings(tree[1].keys())
+    return {k: (reduced[k], reduce_tree(v)) for k, v in tree[1].items()}
+
+
+def generate_aliases(aliases, cfg):
+    """Generate aliases according to command line and level generation."""
+    def process(aliased, what, method):
+        checking = {k: method(f) for k, f in what.items()}
+        if all_unique_values(aliased, checking):
+            return checking
+        return what
+
+    def remove_extension(filename):
+        return filename.rsplit('.', 1)[0]
+
+    def map_file(processed_and_not):
+        # Get the filename and split the extension if we haven't already
+        filename = processed_and_not[0][-1].rsplit('.', 1)
+        mapping = FILE_MAP.get(filename[0], filename[0])
+        # I f we hadn't removed the extension it was because we would be
+        # creating duplicates, so replacing with an alias must also preserve
+        # the extension.
+        if len(filename) == 2:
+            mapping = mapping + '.' + filename[1]
+        return (processed_and_not[0][:-1], mapping)
+
+    def map_dir(processed_and_not):
+        # If there are more path elements try to map the next directory
+        if processed_and_not[0]:
+            elem = processed_and_not[0][-1]
+            if elem in PATH_MAP:
+                return (processed_and_not[0][:-1],
+                        PATH_MAP[elem] + '-' + processed_and_not[1])
+        return processed_and_not
+
+    level = cfg.alias_level
+    if level <= 1:
+        return {p: a or (f if level else p)
+                for p, (f, a, u) in aliases.items()}
```

Comment:
lol single var names are hard to undestand, I'd eventually expand it as we do any future refactor :-)